ros/src/computing/perception/detection/packages/lidar_tracker/nodes/obj_fusion/obj_fusion.cpp

```diff
@@ -56,54 +58,80 @@ static double euclid_distance(const geometry_msgs::Point pos1,
+
+  for (unsigned int i = 0; i < obj_label_msg->obj_id.size(); ++i) {
+    obj_label.reprojected_positions.push_back(
+        obj_label_msg->reprojected_pos.at(i));
+    obj_label.obj_id.push_back(obj_label_msg->obj_id.at(i));
+  }
+
+  std::vector<autoware_msgs::CloudCluster> v_cloud_cluster;
   std_msgs::Header header = sensor_header;
-  std::vector<geometry_msgs::Point> centroids_current;
-
-  LOCK(mtx_reprojected_positions);
-  copy(obj_label.reprojected_positions.begin(),
-       obj_label.reprojected_positions.end(),
-       back_inserter(obj_label_current.reprojected_positions));
-  copy(obj_label.obj_id.begin(), obj_label.obj_id.end(),
-       back_inserter(obj_label_current.obj_id));
-  UNLOCK(mtx_reprojected_positions);
-
-  LOCK(mtx_centroids);
-  copy(centroids.begin(), centroids.end(), back_inserter(centroids_current));
-  copy(v_cloud_cluster.begin(), v_cloud_cluster.end(),
-       back_inserter(v_cloud_cluster_current));
-  UNLOCK(mtx_centroids);
-
-  if (centroids_current.empty() ||
-      obj_label_current.reprojected_positions.empty() ||
-      obj_label_current.obj_id.empty()) {
+  std::vector<geometry_msgs::Point> centroids;
+
+  for (int i(0); i < (int)in_cloud_cluster_array_ptr->clusters.size(); ++i) {
+    autoware_msgs::CloudCluster cloud_cluster =
+        in_cloud_cluster_array_ptr->clusters.at(i);
+    /* convert centroids coodinate from velodyne frame to map frame */
+    tf::Vector3 pt(cloud_cluster.centroid_point.point.x,
+                   cloud_cluster.centroid_point.point.y,
+                   cloud_cluster.centroid_point.point.z);
+    tf::Vector3 converted = tform * pt;
+    sensor_header = cloud_cluster.header;
+    v_cloud_cluster.push_back(cloud_cluster);
+    geometry_msgs::Point point_in_map;
+    point_in_map.x = converted.x();
+    point_in_map.y = converted.y();
+    point_in_map.z = converted.z();
+
+    centroids.push_back(point_in_map);
+  }
+
+  if (centroids.empty() ||
+      obj_label.reprojected_positions.empty() ||
+      obj_label.obj_id.empty()) {
     jsk_recognition_msgs::BoundingBoxArray pub_msg;
     pub_msg.header = header;
     std_msgs::Time time;
     obj_pose_pub.publish(pub_msg);
     autoware_msgs::CloudClusterArray cloud_clusters_msg;
     cloud_clusters_msg.header = header;
     cluster_class_pub.publish(cloud_clusters_msg);
-
+    visualization_msgs::MarkerArray marker_array_msg;
+    marker_array_pub.publish(marker_array_msg);
     time.data = obj_pose_timestamp;
     obj_pose_timestamp_pub.publish(time);
     return;
   }
 
   std::vector<int> obj_indices;
-
-  for (unsigned int i = 0; i < obj_label_current.obj_id.size(); ++i) {
+  for (unsigned int i = 0; i < obj_label.obj_id.size(); ++i) {
     unsigned int min_idx = 0;
     double min_distance = DBL_MAX;
 
     /* calculate each euclid distance between reprojected position and centroids
      */
-    for (unsigned int j = 0; j < centroids_current.size(); j++) {
+    for (unsigned int j = 0; j < centroids.size(); j++) {
```

Comment:
I am not sure why are you using ++i and j++. For for loop either works but looks nicer if consistent.